msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/errors/MFAErrors.kt

```diff
@@ -5,24 +5,47 @@ import com.microsoft.identity.nativeauth.statemachine.results.MFARequiredResult
 import com.microsoft.identity.nativeauth.statemachine.results.MFASubmitChallengeResult
 
 /**
- * MFA error. The user should use the utility methods of this class
- * to identify and handle the error. This error is produced by the various
- * MFA operations.
+ * MFA error. Use the utility methods of this class
+ * to identify and handle the error. This error is produced by
+ * [com.microsoft.identity.nativeauth.statemachine.states.MFARequiredState.requestChallenge] and
+ * [com.microsoft.identity.nativeauth.statemachine.states.AwaitingMFAState.requestChallenge].
+ * @param errorType the error type value of the error that occurred.
+ * @param error the error returned by the authentication server.
+ * @param errorMessage the error message returned by the authentication server.
+ * @param correlationId a unique identifier for the request that can help in diagnostics.
+ * @param errorCodes a list of specific error codes returned by the authentication server.
+ * @param exception an internal unexpected exception that happened.
+ */
+class MFARequestChallengeError(
+    override val errorType: String? = null,
+    override val error: String? = null,
+    override val errorMessage: String?,
+    override val correlationId: String,
+    override val errorCodes: List<Int>? = null,
+    val subError: String? = null,
+    override var exception: Exception? = null
+): MFARequiredResult, BrowserRequiredError, Error(errorType = errorType, error = error, errorMessage= errorMessage, correlationId = correlationId, errorCodes = errorCodes, exception = exception)
+
+/**
+ * MFA error. Use the utility methods of this class
+ * to identify and handle the error. This error is produced by
+ * [com.microsoft.identity.nativeauth.statemachine.states.MFARequiredState.getAuthMethods]
  * @param errorType the error type value of the error that occurred.
  * @param error the error returned by the authentication server.
  * @param errorMessage the error message returned by the authentication server.
  * @param correlationId a unique identifier for the request that can help in diagnostics.
  * @param errorCodes a list of specific error codes returned by the authentication server.
  * @param exception an internal unexpected exception that happened.
  */
-class MFAError(
+class GetAuthMethodsError(
```

Comment:
Should this be "MFAGetAuthMethodsError"

Reply:
I didn't do it as get auth methods won't be specific to MFA (e.g. /introspect might be introduced in the sign up flow). But currently `getAuthMethods()` is on the `MFARequiredState`, so yeah you're right.